### ISSUE TITLE
WinAppDriver not able to identify element but UIRecorder and Accessibility Insights is detecting

### ISSUE THREAD

**Issue by rohit-nehra** [microsoft/WinAppDriver#1526]
Hi, I am trying to automate an application which has different nested panes for controls. I am able to inspect controls using uirecorder and accessibility insights, but for may controls I get "Timeout error" when trying to execute using WinAppDriver.

I captured details for page source just before interacting with control and observed that it doesn't have any elements within parent pane. Attaching screenshots for UIRecorder, Accessibility Insight and Page Source
![AccessibilityInsights](https://user-images.githubusercontent.com/71001539/119435860-62f91a80-bd5e-11eb-89bc-41deee86622b.jpg)
![PageSource](https://user-images.githubusercontent.com/71001539/119435863-655b7480-bd5e-11eb-88ac-423346489ee1.jpg)
![UIRecorder](https://user-images.githubusercontent.com/71001539/119435867-65f40b00-bd5e-11eb-8907-a5efda769544.jpg)

**Comment by anunay1** [comment]
Can you also share the code?

**Comment by rohit-nehra** [comment]
Sure, I will try to post the code as well. Meanwhile I also observed that the controls which are not getting recognized by winappdriver are populated few seconds post screen load. I tried applying static wait, fluent wait as well but still they do not appear in page source.

**Comment by rohit-nehra** [comment]
`var capabilities = new DesiredCapabilities();
 capabilities.setCapability("app", "path to application exe");
WindowsDriver driver = new WindowsDriver(new URL("http://127.0.0.1:4723"),capabilities);
driver.manage().timeouts().implicitlyWait(1, TimeUnit.SECONDS)
driver.findElementByClassName("TextBox").clear();
		driver.findElementByClassName("TextBox").sendKeys("abc");
		driver.findElementByAccessibilityId("ComboBoxDatabases").click();
		driver.findElementByName("abc").click();
		driver.findElementByAccessibilityId("PasswordBoxPassword").sendKeys("abc");
		driver.findElementByName("CONNECT").click();
		Thread.sleep(2000);
		driver.findElementByXPath("//Text[@Name='BRANCH TYPE:']/following::ComboBox").click();
		driver.findElementByName("Over The Counter").click();
		driver.findElementByName("OK").click();
		// switch context
		var allWindowHandles = driver.getWindowHandles();
		var parentHandle = driver.getWindowHandle();
		for (String handle : allWindowHandles) {
			if (handle != parentHandle) {
				this.driver.switchTo().window(handle);
			}
		}
		driver.findElementByName("Customer Search").click();
		driver.findElementByAccessibilityId("TextBoxSearchFor").click();
		driver.findElementByAccessibilityId("TextBoxSearchFor").sendKeys("10002117");
		driver.findElementByAccessibilityId("ButtonSearchOrClear").click();
		driver.findElementByXPath("//Text[@ClassName='TextBlock' and @Name='10002117']").click();
		assertEquals("test user", driver.findElementByXPath("//Edit[@Name='RIM:']").getText()); --- top pane control working
		Thread.sleep(10000);
		System.out.println(driver.getPageSource());
		driver.findElementByXPath(
				"//Pane[@ClassName='ProfUIS-ControlBar'][@Name='Modules Tool Box']//List[@ClassName='ListBox']/ListItem[2]")
				.click(); ---controls in child pane not working`

**Comment by tompahoward** [comment]
same again, but slightly easier to read
```ts
var capabilities = new DesiredCapabilities();
capabilities.setCapability("app", "path to application exe");
WindowsDriver driver = new WindowsDriver(new URL("http://127.0.0.1:4723"),capabilities);
driver.manage().timeouts().implicitlyWait(1, TimeUnit.SECONDS)
driver.findElementByClassName("TextBox").clear();
driver.findElementByClassName("TextBox").sendKeys("abc");
driver.findElementByAccessibilityId("ComboBoxDatabases").click();
driver.findElementByName("abc").click();
driver.findElementByAccessibilityId("PasswordBoxPassword").sendKeys("abc");
driver.findElementByName("CONNECT").click();
Thread.sleep(2000);
driver.findElementByXPath("//Text[@Name='BRANCH TYPE:']/following::ComboBox").click();
driver.findElementByName("Over The Counter").click();
driver.findElementByName("OK").click();
// switch context
var allWindowHandles = driver.getWindowHandles();
var parentHandle = driver.getWindowHandle();
for (String handle : allWindowHandles) {
  if (handle != parentHandle) {
    this.driver.switchTo().window(handle);
  }
}
driver.findElementByName("Customer Search").click();
driver.findElementByAccessibilityId("TextBoxSearchFor").click();
driver.findElementByAccessibilityId("TextBoxSearchFor").sendKeys("10002117");
driver.findElementByAccessibilityId("ButtonSearchOrClear").click();
driver.findElementByXPath("//Text[@ClassName='TextBlock' and @Name='10002117']").click();
assertEquals("test user", driver.findElementByXPath("//Edit[@Name='RIM:']").getText()); --- top pane control working
Thread.sleep(10000);
System.out.println(driver.getPageSource());
driver.findElementByXPath(
    "//Pane[@ClassName='ProfUIS-ControlBar'][@Name='Modules Tool Box']//List[@ClassName='ListBox']/ListItem[2]")
    .click(); // ---controls in child pane not working
```

**Comment by anunay1** [comment]
still can't make out what's happening, please ping me at anunayathakur1@gmail.com

**Comment by tompahoward** [comment]
Is there extra debugging we can turn on?

It's strange that WAD can't see any nodes beneath the `ProfUIS-ControlBar`, but the WAD UI Recorder can see that is has a `Gupta:Dialog`

I'm not sure if it's relevant but when the screen updates after the `driver.findElementByXPath("//Text[@ClassName='TextBlock' and @Name='10002117']").click();` the UI including the `ProfUIS-ControlBar` is displayed, and then a little bit later, the `Gupta:Dialog` then displays.

Is it possible that WAD is not updating the page tree when the `Gupta:Dialog` is added to the tree? Is there a way to tell WAD to rebuild the page tree?

**Comment by anunay1** [comment]
Unfortunately there isn't any extra debugging. if you can show me I can get some idea.

**Comment by rohit-nehra** [comment]
I have dropped you a ping in hangouts at anunayathakur1@gmail.com. Please do let me know when its possible for you to do a screen sharing session today.

**Comment by anunay1** [comment]
@rohit-nehra I did not receive any ping.

**Comment by rohit-nehra** [comment]
@anunay1 , I did send it today morning, it's showing pending invitation. Have sent you gmail as well from my id, alternatively can you drop me a ping on rohit.nehra19@gmail.com

**Comment by rohit-nehra** [comment]
@anunay1 , thanks for your time. I did try with inspect.exe - it doesn't show child controls under parent pane (modules text box pane) in inspect UI tree. Also when I hover mouse on any of these controls I do get properties but none of the elements in UI tree in inspect is highlighted.
<img width="779" alt="Inspect_ChildControl" src="https://user-images.githubusercontent.com/71001539/119921164-e3af5500-bfb0-11eb-8490-e405ab8b980c.png">
<img width="768" alt="Inspect_ParentPane" src="https://user-images.githubusercontent.com/71001539/119921169-e611af00-bfb0-11eb-83d5-f900cd1d2349.png">

**Comment by anunay1** [comment]
That means that there are no automation properties implemented for the controls, if the SW in under your control, please ask the developers to implement that. 

NOTE: of inspect is not able to identify the elements Win App driver will not work.

**Comment by rohit-nehra** [comment]
Thanks for your comments @anunay1 . Appreciate your help !

**Comment by tompahoward** [comment]
So is WAD UI recorder using a different mechanism to build the page tree that WAD?

**Comment by tompahoward** [comment]
Is there a way to offer a bounty for this issue?

**Comment by anunay1** [comment]
I don't think you can get any, but if your list is not dynamic, you can do it in a ugly way, by using keyboard.sendkeys(Keys.DOWN). if keyboard commands are supported.
